Makefile

```diff
@@ -279,53 +299,65 @@ GENOMICSDB_EXAMPLE_SOURCES:= \
 
-ALL_GENOMICSDB_SOURCES := $(GENOMICSDB_LIBRARY_SOURCES) $(GENOMICSDB_EXAMPLE_SOURCES)
+ALL_GENOMICSDB_SOURCES := $(GENOMICSDB_LIBRARY_SOURCES) $(GENOMICSDB_EXAMPLE_SOURCES) $(GENOMICSDB_LIBRARY_TEST_SOURCES)
 
 GENOMICSDB_LIBRARY_OBJ_FILES := $(patsubst %.cc, $(GENOMICSDB_OBJ_DIR)/%.o, $(GENOMICSDB_LIBRARY_SOURCES))
+GENOMICSDB_LIBRARY_TEST_OBJ_FILES := $(patsubst %.cc, $(GENOMICSDB_OBJ_DIR)/%.o, $(GENOMICSDB_LIBRARY_TEST_SOURCES))
 GENOMICSDB_EXAMPLE_OBJ_FILES := $(patsubst %.cc, $(GENOMICSDB_OBJ_DIR)/%.o, $(GENOMICSDB_EXAMPLE_SOURCES))
 GENOMICSDB_EXAMPLE_BIN_FILES := $(patsubst %.cc, $(GENOMICSDB_BIN_DIR)/%, $(GENOMICSDB_EXAMPLE_SOURCES))
 
-ALL_GENOMICSDB_OBJ_FILES:=$(GENOMICSDB_LIBRARY_OBJ_FILES) $(GENOMICSDB_EXAMPLE_OBJ_FILES)
+ALL_GENOMICSDB_OBJ_FILES:=$(GENOMICSDB_LIBRARY_OBJ_FILES) $(GENOMICSDB_EXAMPLE_OBJ_FILES) $(GENOMICSDB_LIBRARY_TEST_OBJ_FILES)
 ALL_GENOMICSDB_HEADER_DEPENDENCIES = $(ALL_GENOMICSDB_OBJ_FILES:%.o=%.d)
 
 GENOMICSDB_STATIC_LIBRARY:=$(GENOMICSDB_BIN_DIR)/libgenomicsdb.a
 GENOMICSDB_SHARED_LIBRARY_BASENAME:=libtiledbgenomicsdb.$(SHARED_LIBRARY_EXTENSION)
 GENOMICSDB_SHARED_LIBRARY:=$(GENOMICSDB_BIN_DIR)/$(GENOMICSDB_SHARED_LIBRARY_BASENAME)
 
+GENOMICSDB_TEST_EXECUTABLE:=$(GENOMICSDB_BIN_DIR)/genomicsdb_test_main
+
 #Put GENOMICSDB_STATIC_LIBRARY as first component of LDFLAGS
 ifeq ($(OS), Darwin)
-    LDFLAGS:=$(GENOMICSDB_BIN_DIR)/libgenomicsdb.a $(LDFLAGS)
+  LDFLAGS:=$(GENOMICSDB_BIN_DIR)/libgenomicsdb.a $(LDFLAGS)
 else
-    LDFLAGS:=-Wl,-Bstatic -L$(GENOMICSDB_BIN_DIR) -lgenomicsdb -Wl,-Bdynamic $(LDFLAGS)
+  LDFLAGS:=-Wl,-Bstatic -L$(GENOMICSDB_BIN_DIR) -lgenomicsdb -Wl,-Bdynamic $(LDFLAGS)
 endif
 
 ###################
 # General Targets #
 ###################
 
 .PHONY: all genomicsdb_library clean clean-dependencies clean-all \
-        TileDB_library TileDB_clean htslib_library htslib_clean
+        TileDB_library TileDB_clean htslib_library htslib_clean \
+        test TileDB_test
 
 ALL_BUILD_TARGETS:= genomicsdb_library
 ifndef DISABLE_MPI
-    ALL_BUILD_TARGETS += $(GENOMICSDB_EXAMPLE_BIN_FILES)
+  ALL_BUILD_TARGETS += $(GENOMICSDB_EXAMPLE_BIN_FILES)
 endif
 ifdef BUILD_JAVA
-    ALL_BUILD_TARGETS += $(GENOMICSDB_JAR)
+  ALL_BUILD_TARGETS += $(GENOMICSDB_JAR)
 endif
 
 all: $(ALL_BUILD_TARGETS)
 
 genomicsdb_library: $(GENOMICSDB_STATIC_LIBRARY) $(GENOMICSDB_SHARED_LIBRARY)
 
 clean:
-	rm -rf $(GENOMICSDB_BIN_DIR)/* $(GENOMICSDB_OBJ_DIR)/*
-	mvn clean -Dgenomicsdb.version=$(RELEASE_VERSION)
+	@rm -rf $(GENOMICSDB_BIN_DIR)/* $(GENOMICSDB_OBJ_DIR)/*
+	@mvn clean -Dgenomicsdb.version=$(RELEASE_VERSION)
+	@rm -rf $(GENOMICSDB_TEST_EXECUTABLE)
 
 clean-dependencies: TileDB_clean htslib_clean
 
 clean-all: clean clean-dependencies
 
+test: genomicsdb_library $(GENOMICSDB_LIBRARY_TEST_OBJ_FILES) $(GENOMICSDB_TEST_EXECUTABLE)
+	@$(GENOMICSDB_TEST_EXECUTABLE)
+	#mvn test -Dgenomicsdb.version=$(RELEASE_VERSION)
+
+$(GENOMICSDB_TEST_EXECUTABLE): $(GENOMICSDB_LIBRARY_TEST_OBJ_FILES)
```

Comment:
I will keep this as a TODO for next PR. I think all additions related to C++ unit tests are not required now